TheraLang.DAL/Repository/Repository.cs

```diff
@@ -9,46 +9,77 @@ namespace TheraLang.DAL.Repository
 {
     public class Repository<TEntity> : IRepository<TEntity> where TEntity : class
     {
+        private DbSet<TEntity> DbSet { get; }
         public Repository(DbSet<TEntity> dbSet)
         {
             DbSet = dbSet;
         }
+        public async Task<TEntity> GetAsync(Expression<Func<TEntity, bool>> where)
+        {
+            var entity = await DbSet.FirstOrDefaultAsync(where);
+            return entity;
+        }
+        public async Task<TEntity> GetAsync()
+        {
+            var entity = await DbSet.FirstOrDefaultAsync();
+            return entity;
+        }
 
-        private DbSet<TEntity> DbSet { get; }
+        public async Task<IQueryable<TEntity>> GetListAsync(Expression<Func<TEntity, bool>> where)
+        {
+            return DbSet.Where(where);
+        }
 
-        public IQueryable<TEntity> Get()
+        public async Task<IQueryable<TEntity>> GetListAsync()
         {
-            return DbSet;
+            return DbSet.AsQueryable();
         }
 
-        public void Remove(TEntity entity)
+        public async Task RemoveAsync(TEntity entity)
         {
-            DbSet.Remove(entity);
+              DbSet.Remove(entity);
         }
 
-        public void RemoveRange(IEnumerable<TEntity> entities)
+        public async Task RemoveRangeAsync(IEnumerable<TEntity> entities)
         {
             DbSet.RemoveRange(entities);
         }
 
-        public Task Add(TEntity entity)
+        public async Task AddAsync(TEntity entity)
         {
-            return DbSet.AddAsync(entity);
+            await DbSet.AddAsync(entity);
         }
 
-        public Task AddRange(IEnumerable<TEntity> entities)
+        public async Task AddRangeAsync(IEnumerable<TEntity> entities)
         {
-            return DbSet.AddRangeAsync(entities);
+            await DbSet.AddRangeAsync(entities);
         }
 
-        public void Update(TEntity entity)
+        public async Task UpdateAsync(TEntity entity)
```

Comment:
No need to use async if not using await
See above for suggestion

Reply:
it does not matter. If you want, I can change it, but, as for me, much more better use one code style for all methods.

Reply:
When you use async you expect method to be async, but it doesn't, so don't you think that this is a bad code style?